[MWAN] Integration guide for Juniper Networks SRX Series Firewalls (#18320)
cloudflare/cloudflare-docs: src/content/docs/magic-wan/configuration/manually/third-party/juniper.mdx

---
pcx_content_type: integration-guide
title: Juniper Networks SRX Series Firewalls

> 

Reviewer: ```suggestion
```

---

This tutorial provides information and examples of how to configure Juniper Networks SRX Series Firewalls with Magic WAN.

## Prerequisites

Confirm that you have the Cloudflare anycast IPs for your account. You should have two IPs allocated.

> The goal is to configure two IPsec tunnels for each endpoint. This provides us with tunnel redundancy and the ability to load balance ingress and egress traffic (via ECMP).

Reviewer: ```suggestion
The goal is to configure two IPsec tunnels for each endpoint. This provides you with tunnel redundancy and the ability to load balance ingress and egress traffic (via ECMP).
```

Additionally, you will need to select two subnets (either `/31` or `/30`) for the Virtual Tunnel Interfaces (`st0.x`) to control what traffic is routed through the tunnels.

## Cloudflare Magic WAN configuration

This section of the document will cover the configuration of:

- Magic IPsec tunnels
- Magic static routes

> ### Magic IPsec Tunnels

Reviewer: ```suggestion
### Magic IPsec tunnels
```

1. Start by [creating the IPsec tunnels](/magic-wan/configuration/manually/how-to/configure-tunnels/#add-tunnels) in the Cloudflare dashboard with the following values:
	- **Tunnel name**: Up to 15 characters (no spaces).
	- **Description** (Optional).

> 	- **Interface address**: This is the Virtual Tunnel Interface (VTI = `st0.x`) RFC1918 address — the IP address specified in this dialog box is the address on the Cloudflare side of the tunnel.

Reviewer: ```suggestion
	- **Interface address**: This is the Virtual Tunnel Interface (VTI = `st0.x`) RFC 1918 address — the IP address specified in this dialog box is the address on the Cloudflare side of the tunnel.
```

	- **Customer endpoint**: This is the public IP address the tunnel will be established with on the Juniper SRX.
	- **Cloudflare endpoint**: One of the two Cloudflare anycast IP addresses.
	- **Pre-shared key**: Choose **Add pre-shared key later**.
2. Select **Add IPsec Tunnel** and fill in the values for the second tunnel to the same Juniper SRX:

> 	- The IP addresses used for the Interface address must be a unique RFC1918 address (`/31` or `/30`).

Reviewer: ```suggestion
	- The IP addresses used for the Interface address must be a unique RFC 1918 address (`/31` or `/30`).
```

	- The **Customer endpoint** is the same IP specified for the first tunnel.
	- The **Cloudflare Endpoint** for the second tunnel will be the second Cloudflare anycast IP provisioned for your account.

> 3. Select **Add tunnels**. We also recommend selecting **Test Tunnels** to ensure that the settings do not conflict with any other tunnels defined in your account and that the correct anycast IP addresses are specified.

Reviewer: ```suggestion
3. Select **Add tunnels**. We also recommend selecting **Test Tunnels** to ensure that the settings do not conflict with any other tunnels defined in your account and that you specified the correct anycast IP addresses.
```

> 4. Because we chose to add a pre-shared key at a later stage, you will see a warning indicator next to the tunnel names after creating them. This is expected behavior and  indicates there is no pre-shared key associated with the tunnel.

Reviewer: ```suggestion
4. Because we chose to add a pre-shared key at a later stage, you will see a warning indicator next to the tunnel names after creating them. This is expected behavior and indicates there is no pre-shared key associated with the tunnel.
```

Cloudflare leverages [equal-cost multi-path routing](/magic-wan/reference/traffic-steering/) to control steering of traffic across the tunnels. The default priority for each route is `100` — traffic will be load-balanced across the two tunnels equally. You can modify the priorities as needed.

1. Create a static route with the following values. Make sure you select the first tunnel in **Tunnel/Next hop**:
	- **Description:** The description for the static route assigned to your first tunnel.
	- **Prefix**: Enter the destination subnet for which this route is intended. For this example, it is `10.1.20.0/24` as stated above.
	- **Tunnel/Next hop**: Choose your first tunnel from the drop-down menu.
	- **Priority**: Default value is `100`.
	- **Region code**: Leave set to **All Regions** unless otherwise specified.
2. Select **Add Static Route** to add a second route for the same subnet. Make sure the second tunnel is selected in **Tunnel/Next hop**.
3. Select **Test routes** to ensure the settings will be accepted, and then select **Add Routes**.
4. Confirm the routes were added correctly in **Magic WAN** > **Configuration** > **Static Routes**.

> ## Juniper SRX Configuration

Reviewer: ```suggestion
## Juniper SRX configuration
```

The configuration settings in this document are based on JUNOS 21.4R3-S4.9.

There may be some differences in the syntax of the commands in the version on your SRX devices, however the principles are the same. Please refer to the Juniper product documentation for more information.

The following elements will be configured on the Juniper SRX firewall(s):

- Add tunnel interfaces (`st0.x`)
- Assign tunnel interfaces to a security zone
- Allow required protocols to both the tunnel and untrust security zones
- IKE configuration
- IPsec configuration
- Static routes
- Security policies

> ### Tunnel Interfaces

Reviewer: ```suggestion
### Tunnel interfaces
```

1. Add two tunnel interfaces:

```txt
set interfaces st0 unit 0 family inet address 10.252.2.21/31
set interfaces st0 unit 1 family inet address 10.252.2.23/31
```

2. Confirm settings:

```txt
admin@srx220> show configuration interfaces st0
```
```txt output
unit 0 {
	family inet {
		address 10.252.2.21/31;
	}
}
unit 1 {
	family inet {
		address 10.252.2.23/31;
	}
}
```

### Security Zone (Cloudflare) - tunnel interfaces

> Define a security zone and add both tunnel interfaces to it. At a minimum, the interfaces should allow ping, but this zone only contains point-to-point connections between the firewall and the customer network namespace. Setting it to `all` for system-services and protocols should be fine.

Reviewer: ```suggestion
Define a security zone and add both tunnel interfaces to it. At a minimum, the interfaces should allow `ping`, but this zone only contains point-to-point connections between the firewall and the customer network namespace. Setting it to `all` for system-services and protocols should be fine.
```
We would use `ping` instead of `all`, and `all` is formatted as monospaced text.

```txt
set security zones security-zone cloudflare interfaces st0.0 host-inbound-traffic system-services all

> set security zones security-zone cloudflare interfaces st0.0 host-inbound-traffic

Reviewer: Not sure if this command should be referencing `st0.1` instead (it appears in the output below).

set security ipsec vpn cf_magic_wan_tun_02 ike no-anti-replay
set security ipsec vpn cf_magic_wan_tun_02 ike ipsec-policy cf_magic_wan_ipsec_pol
set security ipsec vpn cf_magic_wan_tun_02 establish-tunnels immediately
```

### Static routes

This configuration only factors in one local site (`10.1.20.0/24`). In this example, we assume devices in the trust zone need to route traffic to a remote subnet that is at another Magic WAN-protected site (`10.1.100.0/24`).

Define a static route on the SRX to route traffic to `10.1.100.0/24` with redundant routes that reference each of the two tunnels. There are two ways to accomplish this:

**By adding two destinations for the same route:**

```txt
set routing-options static route 10.1.100.0/24 next-hop 10.252.2.20
set routing-options static route 10.1.100.0/24 next-hop 10.252.2.22
```

```txt
admin@srx220> show configuration routing-options static
```
```txt output
route 10.1.100.0/24 next-hop [ 10.252.2.20 10.252.2.22 ];
```

**Or using the [qualified-next-hop](https://www.juniper.net/documentation/us/en/software/junos/static-routing/topics/ref/statement/qualified-next-hop-edit-routing-options.html) option:**

```txt

> admin@srx220> show configuration routing-options | display set

Reviewer: I believe this line should be removed. If we keep it, the next three lines would be output, not commands entered by the user. Can you check, please?

```
```txt output
static {
	route 0.0.0.0/0 next-hop 23.XX.XXX.46;
	route 10.1.100.0/24 {
		next-hop 10.252.2.20;
		qualified-next-hop 10.252.2.22;
	}
}
```

### Security policies

Define security policies to permit traffic flows destined for Magic WAN protected resources. The source/destination zones will need to incorporate the zone containing the tunnel interfaces.

There are two very simple rules to allow traffic bidirectionally — it is generally recommended to start with a similar policy, then to add more stringent rules once general connectivity is established successfully.

**From Cloudflare to *trust*:**

```txt
set security policies from-zone cloudflare to-zone trust policy cloudflare_to_trust match source-address any
set security policies from-zone cloudflare to-zone trust policy cloudflare_to_trust match destination-address any
set security policies from-zone cloudflare to-zone trust policy cloudflare_to_trust match application any
set security policies from-zone cloudflare to-zone trust policy cloudflare_to_trust then permit
set security policies from-zone cloudflare to-zone trust policy cloudflare_to_trust then log session-close
```

```txt

> admin@srx220> show configuration security policies from-zone trust to-zone cloudflare

Reviewer: Should this be `(...) from-zone cloudflare to-zone trust` instead?

```txt
admin@srx220> show security ipsec security-associations
```
```txt output
Total active tunnels: 2
ID    Algorithm       SPI      Life:sec/kb  Mon lsys Port  Gateway
<131073 ESP:aes-cbc-256/sha256 d28e709e 28565/unlim - root 500 162.XXX.66.164
>131073 ESP:aes-cbc-256/sha256 25aed8ae 28565/unlim - root 500 162.XXX.XX.164
<131074 ESP:aes-cbc-256/sha256 3f13176d 28566/unlim - root 500 172.XX.XXX.164
>131074 ESP:aes-cbc-256/sha256 965169e9 28566/unlim - root 500 172.XX.XXX.164
```

### IKE `traceoptions`

It is very helpful to enable debug logging via `traceoptions` while setting up the tunnels. The log data can be exceptionally useful in determining if there are issues and, if so, where they might be occurring.

#### Enable IKE `traceoptions`

```txt
set security ike traceoptions file ike-debug.log
set security ike traceoptions file size 1m
set security ike traceoptions file files 3
set security ike traceoptions file world-readable
set security ike traceoptions flag all
```

The log file can be viewed by doing the following:

1. From an operational mode, run **start shell**.

> 2. Then use the `tail` command to view the contents of the log file in real-time:

Reviewer: ```suggestion
2. Use the `tail` command to view the contents of the log file in real-time:
```
We're already in step 2, so `then` is kind of implied, no?

	```txt
	tail -f /var/log/ike-debug.log
	```

> 3. Press  <kbd>CTRL+C </kbd> when finished.

Reviewer: ```suggestion
3. Press  <kbd>CTRL+C</kbd> when finished.
```
